Classes/Controller/FeuserController.php

```diff
@@ -420,18 +420,13 @@ protected function getFollowingUserGroups(int $currentUserGroup): array
 
     protected function getUserGroupIds(): array
     {
-        $settingsUserGroupKeys = [
-            'usergroup',
-            'usergroupPostSave',
-            'usergroupPostConfirm',
-            'usergroupPostAccept',
-        ];
+        $settingsUserGroupKeys = $this->getUserGroupIdSettingKeys();
```

Comment:
I dont quite get the benefit of the different sorting. As in both cases the result is an array with usergroups.

Reply:
The different sorting takes care of the different usergroups set during the workflow.
This will affect `getFollowingUserGroups()`. If the system is configured in a way, that the Admin Accept step happens before the User's Confirm step `getFollowingUserGroups()` should return a different result.

# Example 1: First User:confirm then Admin:accept

```typoscript
confirmEmailPostCreate = true
acceptEmailPostConfirm = true
```

## Order in which the User groups should be applied:

- usergroupPostSave
- usergroupPostConfirm
- usergroupPostAccept

> This is the default and does work well


# Example 2: First Admin:accept then User:confirm

```typoscript
acceptEmailPostCreate = true
confirmEmailPostAccept = true
```

## Order in which the User groups should be applied:

- usergroupPostSave
- usergroupPostAccept
- usergroupPostConfirm

> In this case it is necessary to adjust the order because otherwise the user will receive a already-registered error on confirmation

## Details

1. User fills in form > User in `usergroupPostSave`
2. Admin accepts user > User in `usergroupPostAccept`
3. User want's to confirm > Error

At step 3. the user is already in the `usergroupPostAccept` group. In the current implementation `getFollowingUserGroups('usergroupPostConfirm')` would return:

- usergroupPostConfirm
- usergroupPostAccept

and the following code will return `TRUE`

```php
$this->isUserInUserGroups(
    $user,
    $this->getFollowingUserGroups((int) $this->settings['usergroupPostConfirm'])
)
```